app/screens/channel_notification_preferences/channel_notification_preferences.tsx

```diff
@@ -0,0 +1,110 @@
+
+import {popTopScreen} from '../navigation';
+
+import MutedBanner, {MUTED_BANNER_HEIGHT} from './muted_banner';
+import NotifyAbout, {BLOCK_TITLE_HEIGHT} from './notify_about';
+import ResetToDefault from './reset';
+import ThreadReplies from './thread_replies';
+
+import type {AvailableScreens} from '@typings/screens/navigation';
+
+type Props = {
+    channelId: string;
+    componentId: AvailableScreens;
+    defaultLevel: NotificationLevel;
+    defaultThreadReplies: 'all' | 'mention';
+    isCRTEnabled: boolean;
+    isMuted: boolean;
+    notifyLevel?: NotificationLevel;
+    notifyThreadReplies?: 'all' | 'mention';
+}
+
+const ChannelNotificationPreferences = ({channelId, componentId, defaultLevel, defaultThreadReplies, isCRTEnabled, isMuted, notifyLevel, notifyThreadReplies}: Props) => {
+    const serverUrl = useServerUrl();
+    const defaultNotificationReplies = defaultThreadReplies === 'all';
+    const diffNotificationLevel = notifyLevel !== 'default' && notifyLevel !== defaultLevel;
+    const notifyTitleTop = useSharedValue((isMuted ? MUTED_BANNER_HEIGHT : 0) + BLOCK_TITLE_HEIGHT);
+    const [notifyAbout, setNotifyAbout] = useState<NotificationLevel>(notifyLevel === undefined || notifyLevel === 'default' ? defaultLevel : notifyLevel);
+    const [threadReplies, setThreadReplies] = useState<boolean>((notifyThreadReplies || defaultThreadReplies) === 'all');
+    const [resetDefaultVisible, setResetDefaultVisible] = useState(diffNotificationLevel || defaultNotificationReplies !== threadReplies);
+
+    useDidUpdate(() => {
+        LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
+    }, [isMuted]);
+
+    const onResetPressed = useCallback(() => {
+        setResetDefaultVisible(false);
+        setNotifyAbout(defaultLevel);
+        setThreadReplies(defaultNotificationReplies);
+    }, [defaultLevel, defaultNotificationReplies]);
+
+    const onNotificationLevel = useCallback((level: NotificationLevel) => {
+        setNotifyAbout(level);
+        setResetDefaultVisible(level !== defaultLevel || defaultNotificationReplies !== threadReplies);
+    }, [defaultLevel, defaultNotificationReplies, threadReplies]);
+
+    const onSetThreadReplies = useCallback((value: boolean) => {
+        setThreadReplies(value);
+        setResetDefaultVisible(defaultNotificationReplies !== value || notifyAbout !== defaultLevel);
+    }, [defaultLevel, defaultNotificationReplies, notifyAbout]);
+
+    const close = () => popTopScreen(componentId);
+
+    const save = useCallback(() => {
+        const pushThreads = threadReplies ? 'all' : 'mention';
+
+        if (notifyLevel !== notifyAbout || (isCRTEnabled && pushThreads !== notifyThreadReplies)) {
+            const props: Partial<ChannelNotifyProps> = {push: notifyAbout};
+            if (isCRTEnabled) {
+                props.push_threads = pushThreads;
+            }
+
+            updateChannelNotifyProps(serverUrl, channelId, props);
+        }
+        close();
```

Comment:
Close is only used here, but instantiated in every render. What about just calling here directly `popTopScreen`?